javascript/javascript3/week2/task_1.js

```diff
@@ -0,0 +1,104 @@
+fetch(
+  "https://gist.githubusercontent.com/pankaj28843/08f397fcea7c760a99206bcb0ae8d0a4/raw/02d8bc9ec9a73e463b13c44df77a87255def5ab9/movies.json"
+)
+  .then((response) => response.json())
+  .then((response) => {
+    console.log(response);
+    const badMovies = response.filter((element) => element.rating < 7);
+    console.log(badMovies);
+    const badMoviesSince = badMovies.filter((element) => element.year >= 2000);
+    console.log(badMoviesSince);
+  });
+
+//creating a promise
+function callMeAsynchronously(resolveAfter) {
+  
+  console.log("starting calling");
+  return new Promise((Resolve) => {
+    setTimeout(() => {
+      Resolve(`is  called asynchronously after ${resolveAfter} seconds`);
+    }, resolveAfter*1000);
+  });
+}
+callMeAsynchronously(5)
+  .then((returnedObject) => {
+    console.log(`the function  ${returnedObject}`);
+  })
+  
+
+//creating a promise with async/await
+async function callMeUsingAwaitAsync() {
+  try {
+    await callMeAsynchronously(5);
+    console.log("I am called asynchronously using async and await function");
+  } catch (error) {
+    console.log(error);
+  }
+}
+callMeUsingAwaitAsync();
+
+
+//rewrite function
+ function geolocation() {
+  const getPositionAsynchronously = true;
+  //console.log("locating your position...");
+  return new Promise((resolve, reject) => {
+    function success(position) {
+      const latitude = position.coords.latitude;
+      const longitude = position.coords.longitude;
+      if (getPositionAsynchronously) {
+        resolve(`latitude:${latitude},longitude:${longitude}`);
+      } else {
+        reject("Unable to retrieve your location");
+      }
+    }
+    navigator.geolocation.getCurrentPosition(success);
+  });
+}
+
+geolocation()
+  .then((Position) => {
+    console.log(`locating your position..${Position}`);
+  })
+  .catch((error) => {
+    console.log(error);
+  });
+
+//api
+//const apiPromise=new Promise((resolve,reject=>{
+
+function toFetchData() {
+  return new Promise((Resolve) => {
+    setTimeout(() => {
+      Resolve();
```

Comment:
Small error here. Function `Resolve` has one argument.
`Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?ts(2794)
lib.es2015.promise.d.ts(33, 34): An argument for 'value' was not provided.`

Reply:
done